scalajslib/test/src/mill/scalajslib/EsModuleRemapTests.scala

```diff
@@ -25,7 +25,10 @@ object EsModuleRemapTests extends TestSuite {
       ESModuleImportMapping.Prefix("@stdlib/linspace", remapTo)
     )
 
-    override lazy val millDiscover = Discover[this.type]
+    override lazy val millDiscover = {
+      import mill.main.TokenReaders.given
```

Comment:
Can this be moved into the `Discover` macro body? The Scala 2 version does `import mill.api.JsonFormatters._;` as part of the macro quasiquote, I assume something similar should be possible in Scala3?

Reply:
no, imports aren't visible to implicit search inside a macro, this is the same problem that was blocking the design of scalasql

Reply:
I will try again maybe if if move where the import is?

Reply:
I think a `import mill.given` at the top of every file would look better, even if it's the same number of lines. For the version in `CodeGen.scala` nobody sees that code so the ugly version is fine.

Is there an open issue for the `imports aren't visible to implicit search inside a macro` thing? If so we should link it somewhere in a comment

Reply:
I did that in the later commits, ill be happy to make that change if the macro itself can't be fixed